kpi/deployment_backends/mock_backend.py

```diff
@@ -260,63 +249,35 @@ def get_submissions(self,
 
         Results can be filtered on instance ids and/or MongoDB filters can be
         passed through `kwargs`
-        TODO support Mongo Query and XML
         """
 
-        submissions = self.get_data('submissions', [])
         kwargs['submission_ids'] = submission_ids
         params = self.validate_submission_list_params(user,
                                                       format_type=format_type,
                                                       **kwargs)
-        permission_filters = params['permission_filters']
-
-        if len(submission_ids) > 0:
-            if format_type == INSTANCE_FORMAT_TYPE_XML:
-                submission_ids = [str(submission_id) for submission_id in
-                                  submission_ids]
-                # ugly way to find matches, but it avoids to load each xml in memory
-                pattern = r'<{id_field}>({submission_ids})<\/{id_field}>'.format(
-                    submission_ids='|'.join(submission_ids),
-                    id_field=self.SUBMISSION_ID_FIELDNAME
-                )
-                submissions = [
-                    submission
-                    for submission in submissions
-                    if re.search(pattern, submission)
-                ]
-            else:
-                submission_ids = [
-                    int(submission_id)
-                    for submission_id in submission_ids
-                ]
-
-                submissions = [
-                    submission
-                    for submission in submissions
-                    if submission.get(self.SUBMISSION_ID_FIELDNAME)
-                    in submission_ids
-                ]
-
-        if permission_filters:
-            submitted_by = [k.get('_submitted_by') for k in permission_filters]
-            if format_type == INSTANCE_FORMAT_TYPE_XML:
-                # TODO handle `submitted_by` too.
-                raise NotImplementedError
-            else:
-                submissions = [
-                    submission
-                    for submission in submissions
-                    if submission.get('_submitted_by') in submitted_by
-                ]
+
+        submissions, total_count = MongoHelper.get_instances(
+            self.mongo_userform_id, **params)
 
         # Python-only attribute used by `kpi.views.v2.data.DataViewSet.list()`
-        self.current_submissions_count = len(submissions)
+        self.current_submissions_count = total_count
+
+        submissions = [
+            MongoHelper.to_readable_dict(submission)
+            for submission in submissions
+        ]
```

Comment:
Unfortunately, lots of tests rely on a list instead of a generator. 
We should open another issue for that?

Reply:
I think it's fine the way it is (or else we would end up storing a second copy of the test submissions somewhere, so that we could refer back to them after the generator was consumed?—sounds like a pain)